src/engine/controls/loopingcontrol.cpp

```diff
@@ -513,17 +513,26 @@ void LoopingControl::setLoopInToCurrentPosition() {
     const mixxx::BeatsPointer pBeats = m_pBeats;
     LoopSamples loopSamples = m_loopSamples.getValue();
     double quantizedBeat = -1;
-    double pos = m_currentSample.getValue();
+
+    SampleOfTrack sampleOfTrack = getSampleOfTrack();
+    // Note: currentPos can be past the end of the track, in the padded
+    // silence of the last buffer. This position might be not reachable in
+    // a future runs, depending on the buffering.
+    double pos = math_min(sampleOfTrack.current, sampleOfTrack.total);
     if (m_pQuantizeEnabled->toBool() && pBeats) {
+        double closestBeat = m_pClosestBeat->get();
         if (m_bAdjustingLoopIn) {
-            double closestBeat = m_pClosestBeat->get();
-            if (closestBeat == m_currentSample.getValue()) {
+            if (closestBeat == pos) {
                 quantizedBeat = closestBeat;
             } else {
                 quantizedBeat = m_pPreviousBeat->get();
             }
         } else {
-            quantizedBeat = m_pClosestBeat->get();
+            if (closestBeat > sampleOfTrack.total) {
+                quantizedBeat = m_pPreviousBeat->get();
```

Comment:
even the previous beat can be after the end, depending on how far the playpos moved past the end.
[QuantizeControl::lookupBeatPositions(double dCurrentSample)](https://github.com/mixxxdj/mixxx/blob/5f5cd28b3f8b471f5f2ca271e1ecacf8f2b1a659/src/engine/controls/quantizecontrol.cpp#L79)